conan_v2_mode.rst

```diff
@@ -0,0 +1,80 @@
+.. _conan_v2_mode:
+
+Road to Conan v2
+================
+
+Conan has started to think about the next major release. We've been gathering feedback
+from the community about our features and we think it's time to break some default
+behaviors, clean the codebase and add space for new developments.
+
+In version 1.23 we have introduced an environment variable that will mimic the behavior
```

Comment:
This is not to mimic behavior, we should be clear. Conan 2.0 will contain very different behavior, that we will not be able to integrate in Conan 1.X codebase. Something like:

"this is to try to minimize the changes in recipes necessary for the future, as well as some configuration and behavior. This does not resemble the full behavior that Conan 2.0 will bring" 

Not exactly this, but something in the line.